### PR TITLE
fix: evict least-active sessions when accept queue is full

### LINKED ISSUE #644: Drop old connections when the accept queue is full
If the accept queue is full, new connections can't be accepted. Could we close the least recently used connections to make some space for those new connections?

Maybe that could be behind a configuration flag. The default would be to close the oldest connections.

In regard of #643 we would still need to send a metric point when this happens.

### LINKED ISSUE #916: worker not restart, (max number of session connection reached, flushing the accept queue)
When generating a lot of traffic at once,
"ERROR max number of session connection reached, flushing the accept queue" and the worker dies.
A dead worker will not be restarted.
Assuming there are 4 workers, if a worker dies one by one and eventually all workers die, sozu will not work.

In my server machine environment
Using apache jmeter, I set the Number of Threads to 2000,
Ramp-up period to 1sec,
Loop Count to Infinite, and set the
When I sent about 2 million requests, the workers started to die in chains.

my config.toml
```toml
log_level = "trace"
log_target = "stdout"

max_connections = 30000
worker_count = 4
worker_automatic_restart = true

[[listeners]]
protocol = "http"
address = "0.0.0.0:7171"

[clusters]
[clusters.myc]
protocol = "http"

frontends = [ { address = "0.0.0.0:7171", hostname = "myhost" } ]

backends = [ { address = "127.0.0.1:7173" } ]
```
linux System resource
```
core file size          (blocks, -c) 0
data seg size           (kbytes, -d) unlimited
scheduling priority             (-e) 0
file size               (blocks, -f) unlimited
pending signals                 (-i) 385941
max locked memory       (kbytes, -l) 64
max memory size         (kbytes, -m) unlimited
open files                      (-n) 65536
pipe size            (512 bytes, -p) 8
POSIX message queues     (bytes, -q) 819200
real-time priority              (-r) 0
stack size              (kbytes, -s) 8192
cpu time               (seconds, -t) unlimited
max user processes              (-u) 385941
virtual memory          (kbytes, -v) unlimited
file locks                      (-x) unlimited
```
error logs
```
WRK-02 ERROR        max number of session connection reached, flushing the accept queue
WRK-03 ERROR        max number of session connection reached, flushing the accept queue
WRK-01 ERROR        max number of session connection reached, flushing the accept queue
WRK-00 ERROR        max number of session connection reached, flushing the accept queue
```

### PR DESCRIPTION
**Pull request by FlorentinDUBOIS**
## Summary

- Add `evict_least_active_sessions()` to `SessionManager` that selects the oldest 1% of non-listener sessions by `last_event()` and closes them when the accept queue is full
- Add `evict_on_queue_full` config option (default: `true`) to `ServerConfig`
- Integrate eviction into `create_sessions()` when connection limits are reached

## Review fixes applied
- Deduplicated session-closing logic by delegating to `shut_down_sessions_by_frontend_tokens`
- Replaced O(n log n) `sort_by_key` with O(n) `select_nth_unstable_by_key` for the hot path
- Fixed unsafe `can_accept = true` bypass → proper `check_limits()` re-validation after eviction
- Added empty-candidates guard to prevent panic on edge case
- Fixed `test_max_connections` e2e test by disabling eviction in test config

Closes #644, closes #916

## Test plan
- [x] `cargo test --workspace` passes
- [x] `cargo clippy --workspace` clean
- [x] E2e tests pass (20/20), including `test_max_connections`

### CODE REVIEW

**Review by llenotre — COMMENTED** [review]
Likely performance issues

File: lib/src/server.rs [hunk 1]
```diff
@@ -1536,6 +1538,54 @@ impl Server {
         gauge!("accept_queue.connections", self.accept_queue.len());
     }
 
+    /// Evict the least recently active sessions to make room for new connections.
+    /// Returns the number of sessions evicted.
+    fn evict_least_active_sessions(&self, count: usize) -> usize {
```
[the remark this answers is not in the document]
Comment (FlorentinDUBOIS) [reply lib/src/server.rs]: You are right, I have updated the code, besides, as it run only when the flag is true, and by default it is false (the defensive choice), I think it should okay.
